predicators/execution_monitoring/expected_atoms_monitor.py

```diff
@@ -30,7 +31,20 @@ def step(self, state: State) -> bool:
         self._curr_plan_timestep += 1
         # If the expected atoms are a subset of the current atoms, then
         # we don't have to replan.
-        unsat_atoms = {a for a in next_expected_atoms if not a.holds(state)}
+        next_expected_vlm_atoms = set(
+            atom for atom in next_expected_atoms
+            if isinstance(atom.predicate, VLMPredicate))
+        non_vlm_unsat_atoms = {
+            a
+            for a in (next_expected_atoms - next_expected_vlm_atoms)
+            if not a.holds(state)
+        }
+        vlm_unsat_atoms = set()
+        if len(next_expected_vlm_atoms) > 0:
+            vlm_unsat_atoms = utils.query_vlm_for_atom_vals(
+                sorted(list(next_expected_vlm_atoms)),
```

Comment:
can we change `query_vlm_for_atom_vals` to take in a Collection of ground atoms, and do the sorting inside that function? from the user's perspective, it's not clear why the input to this function needs to be sorted.

also -- `sorted(next_expected_vlm_atoms)` should work (no `list` needed)